### PR TITLE
new artifacts API

### PR DESCRIPTION
**Pull request by n1ru4l**
TODO
- [x] Basic functionality
- [x] Integration Tests
- [x] Deployment
- [x] Documentation
- [ ] ~~Upgrade client libraries to use the new endpoints~~ (follow-up PR after current changes are live)
- [x] Self-Hosting Guide Update
______

Closes https://github.com/kamilkisiela/graphql-hive/issues/622
Closes https://github.com/kamilkisiela/graphql-hive/issues/621
Closes https://github.com/kamilkisiela/graphql-hive/issues/544

### CODE REVIEW

File: packages/services/cdn-worker/src/artifact-handler.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const ParamsModel = zod.object({
+  targetId: zod.string(),
+  artifactType: zod.union([zod.literal('metadata'), zod.literal('sdl'), zod.literal('services')]),
+});
+
+const authHeaderName = 'x-hive-cdn-key' as const;
+
+export const createArtifactRequestHandler = (deps: ArtifactRequestHandler) => {
+  const router = itty.Router();
```
Comment (ardatan) [inline packages/services/cdn-worker/src/artifact-handler.ts:23]:
```suggestion
  const router = itty.Router<Request & itty.Request>();
```
I think then you can remove the casting below.

Comment (n1ru4l) [reply packages/services/cdn-worker/src/artifact-handler.ts:23]: image

Comment (n1ru4l) [reply packages/services/cdn-worker/src/artifact-handler.ts]: Solved in https://github.com/kamilkisiela/graphql-hive/pull/692/commits/4b292297f5807bb581dcbf54abf2c0474768d291

File: packages/services/api/src/modules/schema/providers/schema-publisher.ts [hunk 2]
```diff
@@ -236,19 +238,22 @@ export class SchemaPublisher {
         includeMetadata: true,
       });
 
+      const orchestrator = this.schemaManager.matchOrchestrator(project.type);
+      const sschemas = schemas.map(s => this.helper.createSchemaObject(s));
```
Comment (kamilkisiela) [inline packages/services/api/src/modules/schema/providers/schema-publisher.ts:242]: sssschemas

File: packages/services/server/src/index.ts [hunk 3]
```diff
@@ -302,6 +333,30 @@ export async function main() {
...
 
+    /** Artifacts API */
+    server.route({
+      method: ['GET'],
+      url: '/artifacts/v1/*',
+      async handler(req, reply) {
+        const response = await artifactRouteHandler.handleNodeRequest(req);
+
+        if (response === undefined) {
+          reply.status(404).send('Not found.'); // eslint-disable-line @typescript-eslint/no-floating-promises -- false positive, FastifyReply.then returns void
```
Comment (kamilkisiela) [inline packages/services/server/src/index.ts:344]:
```suggestion
          void reply.status(404).send('Not found.');
```

**Review by kamilkisiela — APPROVED** [review]

File: packages/services/api/src/modules/schema/providers/schema-publisher.ts (hunk 2 above)
Comment (n1ru4l) [reply packages/services/api/src/modules/schema/providers/schema-publisher.ts]: https://github.com/kamilkisiela/graphql-hive/pull/692/commits/6ea87da73a8465b4ab04ca3512054ae5047d0fbd

**Pull request merged**